Source/CarthageKit/Xcode.swift

```diff
@@ -134,12 +134,28 @@ internal func isSwiftFramework(_ frameworkURL: URL) -> Bool {
 internal func checkSwiftFrameworkCompatibility(_ frameworkURL: URL, usingToolchain toolchain: String?) -> SignalProducer<URL, SwiftVersionError> {
 	return SignalProducer.combineLatest(swiftVersion(usingToolchain: toolchain), frameworkSwiftVersion(frameworkURL))
 		.attemptMap { localSwiftVersion, frameworkSwiftVersion in
-			return localSwiftVersion == frameworkSwiftVersion
+			return localSwiftVersion == frameworkSwiftVersion || isModuleStableAPI(localSwiftVersion, frameworkSwiftVersion, frameworkURL)
 				? .success(frameworkURL)
 				: .failure(.incompatibleFrameworkSwiftVersions(local: localSwiftVersion, framework: frameworkSwiftVersion))
 		}
 }
 
+private func isModuleStableAPI(_ localSwiftVersion: String,
```

Comment:
Would you consider marking this method as `internal` instead of `private` and writing your tests against it instead of relying on resolving the toolchain?

Reply:
I think that is what I’ll have to do. 

Not exactly sure why but I’d prefer an integration test.